projects/extension/ai/openai.py

```diff
@@ -88,3 +88,32 @@ def embed(
         return None
     for obj in response.data:
         yield obj.index, obj.embedding
+
+
+def embed_with_raw_response(
```

Comment:
Just wondering if we could reuse code from the `embed` function or the other way around.
Perhaps, `embed` function could call this new one and just return the parsed response `response.parse()`.